src/components/ContinuationIn/ContinuationAuthorization.vue

```diff
@@ -0,0 +1,393 @@
+
+      <template #action="{ attrs }">
+        <v-btn
+          color="white"
+          icon
+          aria-label="Close Notification"
+          v-bind="attrs"
+          @click="snackbar = false"
+        >
+          <v-icon>mdi-close</v-icon>
+        </v-btn>
+      </template>
+    </v-snackbar>
+  </div>
+</template>
+
+<script lang="ts">
+import { Component, Emit, Mixins, Watch } from 'vue-property-decorator'
+import { Action, Getter } from 'pinia-class'
+import { useStore } from '@/store/store'
+import { DocumentMixin } from '@/mixins'
+import { ContinuationAuthorizationIF, FormIF, PresignedUrlIF } from '@/interfaces'
+import { PdfPageSize } from '@/enums'
+import { VuetifyRuleFunction } from '@/types'
+import FileUploadPreview from '@/components/common/FileUploadPreview.vue'
+import { DatePicker as DatePickerShared } from '@bcrs-shared-components/date-picker'
+// import { LegalServices } from '@/services'
+
+@Component({
+  components: {
+    DatePickerShared,
+    FileUploadPreview
+  }
+})
+export default class ExtraproRegistration extends Mixins(DocumentMixin) {
+  // Refs
+  $refs!: {
+    authorizationDateRef: DatePickerShared,
+    expiryDateRef: DatePickerShared,
+    formRef: FormIF
+  }
+  readonly PdfPageSize = PdfPageSize
+
+  @Getter(useStore) getContinuationAuthorization!: ContinuationAuthorizationIF
+  @Getter(useStore) getCurrentDate!: string
+  @Getter(useStore) getKeycloakGuid!: string
+  @Getter(useStore) getShowErrors!: boolean
+
+  @Action(useStore) setContinuationAuthorization!: (x: ContinuationAuthorizationIF) => void
+
+  readonly INPUT_FILE_LABEL = 'Continuation authorization'
+
+  // Local properties
+  authorization = null as ContinuationAuthorizationIF
+  fileValidity = false
+  fileUploadCustomErrorMsg = ''
+  snackbar = false
+  snackbarText = ''
+
+  get authorityNameRules (): Array<VuetifyRuleFunction> {
+    return [
+      (v: string) => !!v || 'Authority Name is required'
+    ]
+  }
+
+  get authorizationDateRules (): Array<VuetifyRuleFunction> {
+    return [
+      (v) => !!v || 'Authorization Date is required',
+      () => (this.authorization.date <= this.getCurrentDate) ||
+        'Authorization Date cannot be in the future'
+    ]
+  }
+
+  get expiryDateRules (): Array<VuetifyRuleFunction> {
+    return [
+      () => !this.authorization.expiryDate ||
+        (this.authorization.expiryDate >= this.authorization.date) ||
+        'Expiry Date cannot be before Authorization Date'
+    ]
+  }
+
+  /** The number of file upload componenets to display (max 5). */
+  get numUploads (): number {
+    return Math.min((this.authorization.files.length + 1), 5)
+  }
+
+  mounted (): void {
+    // set or initialize authorization object
+    this.authorization = this.getContinuationAuthorization || { files: [] } as ContinuationAuthorizationIF
+  }
+
+  /**
+   * Called when FileUploadPreview tells us whether a file is valid.
+   * This is called right before the File Selected event.
+   */
```

Comment:
This is another trick to using the file upload component: knowing that the file-validity event comes right before the file-selected event.

Since there's only 1 file uploaded at a time, I simply set a variable (line 263) and then check it (line 276). No need for an array.